### PR TITLE
fix(secu) remove unused params or sanitize them in service by hostgroup

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

Security fix which remove unused http parameters (eg: $_GET) or sanitize them in service by hostgroup
**Fixes** # (none)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

please contact me

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [x] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [x] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php [hunk 1]
```diff
@@ -53,18 +53,26 @@
...
+/*
+ *  Check Arguments from GET and session
+ */
+// integer values from $_GET
+$p = filter_input(INPUT_GET, 'p', FILTER_VALIDATE_INT, array('options' => array('default' => 2)));
+$num = filter_input(INPUT_GET, 'num', FILTER_VALIDATE_INT, array('options' => array('default' => 0)));
+$limit = filter_input(INPUT_GET, 'limit', FILTER_VALIDATE_INT, array('options' => array('default' => 20)));
+
+// use whitelist values instead of the $_GET value
+$order = isset($_GET['order']) && $_GET['order'] === "DESC" ? "DESC" : "ASC";
```
Comment (kduret) [inline www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php]: you could use filter_input with FILTER_VALIDATE_REGEXP flag

File: www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php [hunk 2]
```diff
@@ -53,18 +53,26 @@
...
-$limit = $obj->checkArgument("limit", $_GET, 20);
-$instance = $obj->checkArgument("instance", $_GET, $obj->defaultPoller);
-$hostgroups = $obj->checkArgument("hostgroups", $_GET, $obj->defaultHostgroups);
 $search = $obj->checkArgument("search", $_GET, "");
 $sort_type = $obj->checkArgument("sort_type", $_GET, "host_name");
-$order = $obj->checkArgument("order", $_GET, "ASC");
-$dateFormat = $obj->checkArgument("date_time_format_status", $_GET, "Y/m/d H:i:s");
+
+// values saved in the session
+$instance = filter_var($obj->defaultPoller, FILTER_VALIDATE_INT);
```
Comment (kduret) [inline www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php]: I think poller filter does not work anymore, cause you use always session var

File: www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php [hunk 3]
```diff
@@ -53,18 +53,26 @@
...
+$hostgroup = filter_var($obj->defaultHostgroups, FILTER_VALIDATE_INT);
```
Comment (kduret) [inline www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php]: I think hostgroup filter does not work anymore, cause you use always session var

File: www/include/monitoring/status/ServicesHostGroups/xml/serviceSummaryByHGXML.php [hunk 4]
```diff
@@ -53,18 +53,26 @@
...
-$instance = $obj->checkArgument("instance", $_GET, $obj->defaultPoller);
-$hostgroups = $obj->checkArgument("hostgroups", $_GET, $obj->defaultHostgroups);
 $search = $obj->checkArgument("search", $_GET, "");
 $sort_type = $obj->checkArgument("sort_type", $_GET, "alias");
-$order = $obj->checkArgument("order", $_GET, "ASC");
-$dateFormat = $obj->checkArgument("date_time_format_status", $_GET, "Y/m/d H:i:s");
+
+// values saved in the session
+$instance = filter_var($obj->defaultPoller, FILTER_VALIDATE_INT);
+$hostgroup = filter_var($obj->defaultHostgroups, FILTER_VALIDATE_INT);
```
Comment (kduret) [inline www/include/monitoring/status/ServicesHostGroups/xml/serviceSummaryByHGXML.php]: I think hostgroup filter does not work anymore, cause you use always session var

File: www/include/monitoring/status/ServicesHostGroups/xml/serviceSummaryByHGXML.php [hunk 5]
```diff
@@ -53,18 +53,26 @@
...
-$limit = $obj->checkArgument("limit", $_GET, 20);
-$instance = $obj->checkArgument("instance", $_GET, $obj->defaultPoller);
-$hostgroups = $obj->checkArgument("hostgroups", $_GET, $obj->defaultHostgroups);
 $search = $obj->checkArgument("search", $_GET, "");
 $sort_type = $obj->checkArgument("sort_type", $_GET, "alias");
-$order = $obj->checkArgument("order", $_GET, "ASC");
-$dateFormat = $obj->checkArgument("date_time_format_status", $_GET, "Y/m/d H:i:s");
+
+// values saved in the session
+$instance = filter_var($obj->defaultPoller, FILTER_VALIDATE_INT);
```
Comment (kduret) [inline www/include/monitoring/status/ServicesHostGroups/xml/serviceSummaryByHGXML.php]: I think poller filter does not work anymore, cause you use always session var

File: www/include/monitoring/status/ServicesHostGroups/xml/serviceSummaryByHGXML.php [hunk 6]
```diff
@@ -53,18 +53,26 @@
...
+/*
+ *  Check Arguments from GET and session
+ */
+// integer values from $_GET
+$p = filter_input(INPUT_GET, 'p', FILTER_VALIDATE_INT, array('options' => array('default' => 2)));
+$num = filter_input(INPUT_GET, 'num', FILTER_VALIDATE_INT, array('options' => array('default' => 0)));
+$limit = filter_input(INPUT_GET, 'limit', FILTER_VALIDATE_INT, array('options' => array('default' => 20)));
+
+// use whitelist values instead of the $_GET value
+$order = isset($_GET['order']) && $_GET['order'] === "DESC" ? "DESC" : "ASC";
```
Comment (kduret) [inline www/include/monitoring/status/ServicesHostGroups/xml/serviceSummaryByHGXML.php]: you could use filter_input with FILTER_VALIDATE_REGEXP flag

File: www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php (hunk 2 above)
Comment (sc979) [reply www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php]: the instance and hostgroups parameters are not sent in the GET request.
but they're usable to to overload the $instance and $hostgroups parameters

File: www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php (hunk 1 above)
Comment (sc979) [reply www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php]: wouldn't a regexp consume more ressource than a single whitelist string check ?
as there's only two possibilities, and by default ASC should be set

Comment (kduret) [reply www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php]: a simple regex does not consume lot of resource
It's not for this kind of stuff which we need to look at performance issues

Comment (sc979) [reply www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php]: done

**Pull request merged**